code-challenges.test.js

```diff
@@ -20,25 +20,58 @@ var colors1 = ["purple", "blue", "green", "yellow", "pink"]
 var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
-
+describe("shuffleColors", () => {
+  it("takes in an array, removes the first item from the array and shuffles the remaining content", () => {
+    expect(shuffleColors(colors1)).toContain("yellow" && "blue" && "pink" && "green")
+    expect(shuffleColors(colors2)).toContain("saffron" && "aquamarine" && "periwinkle" && "indigo" && "ochre")
+  })
+})
 
 // b) Create the function that makes the test pass.
 
+//First I want to define a function that takes in an array
+const shuffleColors = (array) => {
+  //Next I want to cut the first index off of the input array. I will not need to define a new variable because .shift is a mutator that will permanently alter the input array
+  array.shift()
+  //Finally I want to randomize the indeces of the remaining array. I had to do research but found a way to randomize using the sort function.
+    //.sort iterates over an array evaluating if the current value minus the next value is less than zero, if so, it evaluates to true and places the value at the beginning of the array until all values in the array meet this condition. 
+      //Usually .sort applied to an array of strings orders them alphabetically, but that evaluation is moot if we set the method to check numbers instead.
+      //Math.random generates a float between 0 and 1. If we set the .sort method to compare it against the exact middle point of this range (0.5), the method will evaluate to true (or the first value minus the second value is less than zero) any time the random float is above 0.5. Any time the evaluation randomly evaluates to true, it will log the element of our input array at that iteration to the beginning of the new array until all values are true.
+      //Sorry for the essay but I really wanted to know how this works. 
+  return array.sort((a, b) => 0.5 - Math.random())
+}
```

Comment:
This is exactly what I am looking for! Great essay: Sort is comparing a and b using (some kind of) parameter defined by us as devs. a - b creates a bubble sort and your +/- creates an array where you are swapping the places of the indexes half the time. Its simple and useful though not extremely "RANDOM." For instance its unlikely to get an index that moves more than 4 spaces over (think flipping a coins heads up 4 times) but is totally possible. 

unless of course I am understanding it wrong, which is always an option.